### ISSUE TITLE
React Fragments and components do not output any text

### ISSUE THREAD

**Issue by astorije** [airbnb/enzyme#2028]
This is a follow-up of https://github.com/airbnb/enzyme/issues/1799#issuecomment-468099145 after @ljharb rightfully mentioned creating a new issue.
The issue is a little confusing to me, and I'm not sure I titled it too well, so feel free to re-title it.

### Current behavior

Assuming the following test:

```tsx
describe('<Foobar />', () => {
  test('should include Foo', () => {
    const wrapper = mount(<Foobar />);
    expect(wrapper.text()).toContain('Foo');
  });
});
```

and the following helper component, `Title`:

```tsx
import React from 'react';
const Title = ({ title }: { title: string }) => <span>{title}</span>;
```

The test fails when running against the following component:

```tsx
// ❌ Fails
const Foobar = () => (
  <>
    <Title title="Foo" />
    <>Bar</>
  </>
);
```

`wrapper.text()` does not contain anything:

```
  ● <Foobar /> › should include Foo
    expect(string).toContain(value) // indexOf
    Expected value:  "Foo"
    Received string: ""
```

What is bizarre to me is that all the following derivatives of the `Foobar` component (one small change at a time) **pass** the same test:

```tsx
// ✅ Passes when `Title` is the only child of the fragment
const Foobar = () => (
  <>
    <Title title="Foo" />
  </>
);

// ✅ Passes when the wrapping element is a <div> and not a <> fragment
const Foobar = () => (
  <div>
    <Title title="Foo" />
    <>Bar</>
  </div>
);

// ✅ Passes when using a <span> element instead of the <Title> component
const Foobar = () => (
  <>
    <span>Foo</span>
    <>Bar</>
  </>
);
```

### Expected behavior

The failing test should pass just like the others.

#### API

- [x] mount

#### Version

| library             | version
| ------------------- | -------
| enzyme              | 3.9.0
| react               | 16.8.3
| react-dom           | 16.8.3
| react-test-renderer | 16.6.0
| adapter (below)     | 1.10.0

#### Adapter

- [x] enzyme-adapter-react-16

**Comment by ljharb** [comment]
@astorije react-test-renderer must always have the exact same minor as react and react-dom; can you try updating it to 16.8?

**Comment by ljharb** [comment]
(I'm able to reproduce this regardless, so that's probably a red herring - good to fix anyways tho)

**Comment by astorije** [comment]
@ljharb, thanks for looking into this! We don't pull `react-test-renderer` ourselves, it comes from the adapter. In our `yarn.lock`:

```
enzyme-adapter-react-16@1.10.0:
  version "1.10.0"
  resolved "https://registry.yarnpkg.com/enzyme-adapter-react-16/-/enzyme-adapter-react-16-1.10.0.tgz#12e5b6f4be84f9a2ef374acc2555f829f351fc6e"
  integrity sha512-0QqwEZcBv1xEEla+a3H7FMci+y4ybLia9cZzsdIrId7qcig4MK0kqqf6iiCILH1lsKS6c6AVqL3wGPhCevv5aQ==
  dependencies:
    enzyme-adapter-utils "^1.10.0"
    object.assign "^4.1.0"
    object.values "^1.1.0"
    prop-types "^15.6.2"
    react-is "^16.7.0"
    react-test-renderer "^16.0.0-0"

[...]

react-test-renderer@^16.0.0-0:
  version "16.4.1"
  resolved "https://registry.yarnpkg.com/react-test-renderer/-/react-test-renderer-16.4.1.tgz#f2fb30c2c7b517db6e5b10ed20bb6b0a7ccd8d70"
  dependencies:
    fbjs "^0.8.16"
    object-assign "^4.1.1"
    prop-types "^15.6.0"
    react-is "^16.4.1"
```

I can see that `react-test-renderer` is not a dependency of the adapter anymore, and I couldn't find anywhere mention of having to keep them in sync / having to explicitly have `react-test-renderer` in `package.json`, did I miss anything?

Now that you have fixed the issue in https://github.com/airbnb/enzyme/commit/8bbb61d336bf35a20ce7742824f91f2b8faf641f, is the resolution to wait for a new version bump of the adapter(s), or is there anything else to do?

Thanks again!

**Comment by ljharb** [comment]
If you have a lockfile, you'll have to manually update the transitive dependency so that the minor versions match - or, easier, you can explicitly depend on `react-test-renderer` at the top, and npm/yarn will dedupe it.

https://github.com/airbnb/enzyme/commit/8bbb61d336bf35a20ce7742824f91f2b8faf641f is released in `enzyme-adapter-utils` v1.10.1.
